src/main.js

```diff
@@ -95,19 +96,113 @@ buttons.forEach((button) => {
   });
 });
 
-//shrink header on scroll
+// Object Literal Module
```

Comment:
You were almost there Maya! 

So, the problem you were having here is that you forgot the `this`, otherwise the fn does not exist.
```
this.visited.addEventListener("click", addVisited);
this.visited.addEventListener("click", this.addVisited);
```

As I mentioned last week, you can create a module without using `this`. One of the advantages is this type of error you add and many other tricky scenarios where dealing with `this` can be a nice struggle. Nevertheless I recommend you to keep practice `this` in a separate codepen :)

Here's the code working (put it in the top of the file after imports) and remove the forEach you did in line 10 (I included it in the code example)

<details>

```js
// Object Literal Module
(function () {
  // Set the "shared" variables among different methods:
  let buttons;
  let visited;
  let bucketlist;
  let all;
  let gallery;

  function cacheDom() {
    buttons = Array.from(document.querySelector(".buttons").children);
    visited = buttons[0];
    bucketlist = buttons[1];
    all = buttons[2];
    gallery = document.querySelector(".gallery");
  }

  function bindEvents() {
    visited.addEventListener("click", addVisited);
    bucketlist.addEventListener("click", addBucketlist);
    all.addEventListener("click", showAll);
  }

  function renderDestination(destination) {
    // create destination wrapper
    const elDestination = document.createElement("figure");
    elDestination.classList.add("img-container");

    // add the correct class to elDestination wrapper
    if (destination.visited === true) {
      elDestination.classList.add("visited");
    } else {
      elDestination.classList.add("bucketlist");
    }

    // create img element
    const img = document.createElement("img");
    img.style.objectFit = "cover"; // TODO - Prefer adding a CSS class instead.
    img.src = destination.photo;
    // add img to destination
    elDestination.appendChild(img);

    //set img captions
    const figcaption = document.createElement("figcaption");
    figcaption.classList.add("caption");
    figcaption.insertAdjacentHTML(
      "beforeend",
      `<h2 class='caption__city'>${destination.city}</h2>`
    );
    figcaption.insertAdjacentHTML(
      "beforeend",
      `<h3 class='caption__country'>${destination.country}</h3>`
    );
    // add captions to destination
    elDestination.appendChild(figcaption);

    // Return the destination HTML
    return elDestination;
  }

  function render() {
    // Create DOM elements for each destination
    destinationsList.forEach((destination) => {
      const elDestination = renderDestination(destination);
      gallery.appendChild(elDestination);
    });
  }

  function addVisited() {
    gallery.classList.add("visited");
    gallery.classList.remove("bucketlist");
  }

  function addBucketlist() {
    gallery.classList.add("bucketlist");
    gallery.classList.remove("visited");
  }

  function showAll() {
    gallery.classList.remove("visited");
    gallery.classList.remove("bucketlist");
  }

  function init() {
    cacheDom();
    bindEvents();
    render();
  }

  init();
})();
```

</details>

Reply:
After spending hours trying to refactor the first part of the code I was amazed coming back to your solution and seeing how simple it can be and how complicated mine got (I left it there for comparison) 😄. The hardest part of it was figuring out how different sections can communicate and I have a much clearer picture now. I added the remaining few functions that manipulate the header just for practice but I'm guessing it should become a separate module as the project grows?

Reply:
Regarding this comment: 
```    img.style.objectFit = "cover"; // TODO - Prefer adding a CSS class instead.```
I'm not sure what you mean? Add a class with background-size: cover?

Reply:
> figuring out how different sections can communicate and I have a much clearer picture now.

When building frontend websites, that's a very common challenge, every app is slightly different. You get used to it as you go :)

> I'm not sure what you mean? Add a class with background-size: cover?

Exactly!